haystack/components/extractors/llm_metadata_extractor.py

```diff
@@ -13,50 +13,15 @@
 
 from haystack import Document, component, default_from_dict, default_to_dict, logging
 from haystack.components.builders import PromptBuilder
-from haystack.components.generators.chat import AzureOpenAIChatGenerator, OpenAIChatGenerator
```

Comment:
In this module, you can find how we could simplify the `LLMMetadataExtractor`.
This is just an example, and we should respect our breaking changes policy, first introducing `chat_generator` (in 2.x.0 release) and then removing `generator_api` and `generator_api_params` (in 2.x+1.0 release).